src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  health = 100;
+  hidden = false;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast.hidden === false && beast instanceof Herbivore === true) {
+      beast.health -= 50;
+
+      for (let i = 0; i < Animal.alive.length; i++) {
+        if (Animal.alive[i].health === 0) {
+          delete Animal.alive[i];
+        }
+      }
```

Comment:
it's better not to change static properties outside of the Animal class. What about creating a specific static method for that?
and there is no need to use for loop here. you can always check the `beast` you've gotten in `bite` method.